### PR TITLE
Clients count in statusbar

### PR DESCRIPTION
**Pull request by IKrukov-HORIS**

### CODE REVIEW

File: projector-server/src/main/kotlin/org/jetbrains/projector/server/ProjectorServer.kt [hunk 1]
```diff
@@ -155,6 +158,7 @@ class ProjectorServer private constructor(
 
     builder.onWsClose = { connection ->
       // todo: we need more informative message, add parameters to this method inside the superclass
+      clientsCount = maxOf(0, clientsCount - 1)
```
Comment (SerVB) [inline projector-server/src/main/kotlin/org/jetbrains/projector/server/ProjectorServer.kt:161]: Strictly speaking, modifications of this property aren't thread-safe now. They should be synchronized or atomic

File: projector-plugin/src/main/kotlin/org/jetbrains/projector/plugin/ui/ProjectorStatusWidget.kt [hunk 2]
```diff
@@ -125,12 +143,25 @@ class ProjectorStatusWidget(project: Project)
...
-    val ID : String by lazy { ProjectorStatusWidget::class.java.name }
+    val ID: String by lazy { ProjectorStatusWidget::class.java.name }
     private fun getIcon(path: String): Icon = IconLoader.getIcon(path, ProjectorStatusWidget::class.java)
     private val ACTIVATION_NEEDED_SIGN: Icon by lazy { getIcon("/META-INF/activationNeeded.svg") }
     private val RUNNING_SIGN: Icon by lazy { getIcon("/META-INF/runningSign.svg") }
     private val STARTING_SIGN: Icon by lazy { getIcon("/META-INF/startingSign.svg") }
     private val DISABLED_SIGN: Icon by lazy { getIcon("/META-INF/disabledSign.svg") }
   }
+
+  override fun update(p0: Observable?, p1: Any?) {
```
Comment (SerVB) [inline projector-plugin/src/main/kotlin/org/jetbrains/projector/plugin/ui/ProjectorStatusWidget.kt:161]: Let's use more meaningful argument names, at least `o` and `arg` like in [javadoc](https://docs.oracle.com/javase/8/docs/api/java/util/Observer.html#update-java.util.Observable-java.lang.Object-)

File: projector-agent/src/main/kotlin/org/jetbrains/projector/agent/AgentLauncher.kt [hunk 3]
```diff
@@ -102,6 +104,26 @@ public object AgentLauncher {
     disconnectByIpMethod?.invoke(null, ip)
   }
 
+  @JvmStatic
+  @Suppress("deprecation")
```
Comment (SerVB) [inline projector-agent/src/main/kotlin/org/jetbrains/projector/agent/AgentLauncher.kt:108]: I would vote against using deprecated stuff... I see there are some suggestions on replacement in Observable class, have you considered them?

File: projector-server/src/main/kotlin/org/jetbrains/projector/server/ProjectorServer.kt [hunk 4]
```diff
@@ -168,6 +172,7 @@ class ProjectorServer private constructor(
     }
 
     builder.onWsOpen = { connection ->
+      clientsCount += 1
```
Comment (SerVB) [inline projector-server/src/main/kotlin/org/jetbrains/projector/server/ProjectorServer.kt:175]: Fmpov, it's too early: we should increment the counter only for the clients that have ReadyClientSettings (so they passed handshake well, including correct password and positive connection confirmation).

**Review by SerVB — CHANGES_REQUESTED** [review]

File: projector-server/src/main/kotlin/org/jetbrains/projector/server/ProjectorServer.kt (hunk 4 above)
Comment (IKrukov-HORIS) [reply projector-server/src/main/kotlin/org/jetbrains/projector/server/ProjectorServer.kt]: Done

**Review by SerVB — APPROVED** [review]

**Pull request merged**